ed25519/verifier.go

```diff
@@ -0,0 +1,124 @@
+package ed25519
+
+import (
+	"context"
+	"crypto/ed25519"
+	"crypto/x509"
+	"encoding/pem"
+	"fmt"
+	"strings"
+
+	wrapping "github.com/hashicorp/go-kms-wrapping/v2"
+	"golang.org/x/exp/slices"
+)
+
+// Verifier provides and ed25519 implementation for the wrapping.Verifier interface
+type Verifier struct {
+	pubKey      ed25519.PublicKey
+	keyId       string
+	keyPurposes []wrapping.KeyPurpose
+	keyType     wrapping.KeyType
+}
+
+var _ wrapping.SigInfoVerifier = (*Verifier)(nil)
+var _ wrapping.KeyExporter = (*Verifier)(nil)
+
+// NewVerifier creates a new verifier.  Supported options: WithKeyId,
+// WithKeyPurposes, WithPubKey, WithConfigMap
+//
+// Note: options order of precedence is: local options, WithConfigMap provided
+// options and finally wrapping options.
+func NewVerifier(ctx context.Context, opt ...wrapping.Option) (*Verifier, error) {
+	const op = "ed25519.NewEd25519Verifier"
+	opts, err := getOpts(opt...)
+	if err != nil {
+		return nil, fmt.Errorf("%s: %w", op, err)
+	}
+	if len(opts.WithKeyPurposes) == 0 {
+		opts.WithKeyPurposes = append(opts.WithKeyPurposes, wrapping.KeyPurpose_Verify)
+	}
+	return &Verifier{
+		pubKey:      opts.WithPubKey,
+		keyId:       opts.WithKeyId,
+		keyPurposes: opts.WithKeyPurposes,
+		keyType:     wrapping.KeyType_ED25519,
+	}, nil
+}
+
+// SetConfig sets the fields on the Signer
```

Comment:
You say "Signer" a few times in the comment and in the `op`.

Reply:
ty!